arcgis-ios-sdk-samples/Maps/Display device location with NMEA data sources/DisplayDeviceLocationWithNMEADataSourcesViewController.swift

```diff
@@ -24,31 +24,130 @@ class DisplayDeviceLocationWithNMEADataSourcesViewController: UIViewController {
             mapView.map = AGSMap(basemapStyle: .arcGISNavigation)
         }
     }
+    /// The label to display accuracy info.
+    @IBOutlet var accuracyStatusLabel: UILabel!
     /// The label to display satellites info.
-    @IBOutlet var statusLabel: UILabel!
-    /// The button to start the demo.
-    @IBOutlet var startBarButtonItem: UIBarButtonItem!
+    @IBOutlet var satelliteStatusLabel: UILabel!
+    /// The button to choose a data source and start the demo.
+    @IBOutlet var sourceBarButtonItem: UIBarButtonItem!
     /// The button to reset pan mode to "recenter".
     @IBOutlet var recenterBarButtonItem: UIBarButtonItem!
     /// The button to reset the demo.
     @IBOutlet var resetBarButtonItem: UIBarButtonItem!
     
+    // MARK: Constants
+    
+    /// The protocols used in this sample to get NMEA sentences.
+    /// They are also specified in the `Info.plist` to allow the app to
+    /// communicate with external accessory hardware.
+    let supportedProtocolStrings = [
+        "com.bad-elf.gps",
+        "com.eos-gnss.positioningsource",
+        "com.geneq.sxbluegpssource"
+    ]
+    
     // MARK: Instance properties
     
     /// An NMEA location data source, to parse NMEA data.
-    let nmeaLocationDataSource = AGSNMEALocationDataSource(receiverSpatialReference: .wgs84())
+    var nmeaLocationDataSource: AGSNMEALocationDataSource!
     /// A mock data source to read NMEA sentences from a local file, and generate
     /// mock NMEA data every fixed amount of time.
     let mockNMEADataSource = SimulatedNMEADataSource(nmeaSourceFile: Bundle.main.url(forResource: "Redlands", withExtension: "nmea")!, speed: 1.5)
+    /// A formatter for the accuracy distance string.
+    let distanceFormatter: MeasurementFormatter = {
+        let formatter = MeasurementFormatter()
+        formatter.unitOptions = .naturalScale
+        formatter.numberFormatter.minimumFractionDigits = 1
+        formatter.numberFormatter.maximumFractionDigits = 1
+        return formatter
+    }()
     
     // MARK: Actions
     
-    @IBAction func start() {
-        // Set buttons states.
-        startBarButtonItem.isEnabled = false
-        resetBarButtonItem.isEnabled = true
+    /// Get the first connected and supported Bluetooth accessory with its
+    /// protocol string.
+    /// - Returns: A tuple of the accessory and its protocol,
+    ///            or nil if no supported accessory exists.
+    func firstSupportedAccessoryWithProtocol() -> (EAAccessory, String)? {
+        for accessory in EAAccessoryManager.shared().connectedAccessories {
+            // The protocol string to establish the EASession.
+            guard let protocolString = accessory.protocolStrings.first(where: { supportedProtocolStrings.contains($0) }) else {
```

Comment:
Assuming that `supportedProtocolString` is ordered from best to worst:

```suggestion
            guard let protocolString = supportedProtocolStrings.first(where: { accessory.protocolStrings.contains($0) }) else {
```

Reply:
Actually I don't know what is the best order here, so I'll probably apply your suggestion. 🤔  It doesn't matter for most of the time, as we only expected 1 supported accessory connected each time.

Since I moved to hardcode the protocol strings in the sample, it is guaranteed that each type of NMEA accessory will only have 1 protocol for this sample. So it doesn't involve the issue in Apple doc for `EAAccessory.protocolStrings`

> If your application supports multiple protocols for a single accessory, your code should always choose the highest-fidelity protocol that you support.

- Using `accessory.protocolStrings.first(where: { supportedProtocolStrings.contains($0) })`

Will use the first available connected accessory, ordered by the order of `EAAccessoryManager.shared().connectedAccessories` (probably least recently connected? I don't know)

- Using `supportedProtocolStrings.first(where: { accessory.protocolStrings.contains($0) })`

Will use the first accessory ordered by brand: BadElf, EOS, SXBlue.